cogs/autopaste.py

```diff
@@ -64,11 +64,17 @@ async def interaction_check(self, interaction: Interaction) -> bool:
             await self.on_timeout()
             return False
 
-        if (
-            interaction.user.id == self.message_author.id
-            or interaction.channel.permissions_for(interaction.user).manage_messages
-        ):  # type: ignore
-            return True
+        if isinstance(interaction.channel, Thread) and interaction.channel.parent_id == HELP_CHANNEL_ID:
+            if interaction.user.get_role(HELP_MOD_ID):
```

Comment:
should apply the manage message check before the thread author as well.

Reply:
reworked the check to always allow users with manage_messages. thanks